### PR TITLE
:bug: Parsing an enum response fails on iOS 12

### PR DESCRIPTION
**Pull request by MadsBogeskov**
If you had a response with a type that was just an enum the parser would fail on iOS 12. The reason is that JSONDecoder doesn't support JSON fragments which an enum would be (it is sent as something like this: `"\"ENUM_VALUE\”\n"`)

This fix changes it to be a custom string parser, which then converts it into the right enum value.

### CODE REVIEW

**Review by mxsc — APPROVED** [review]

File: Sources/SwaggerSwiftCore/Utilities/String+IsNumber.swift [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import Foundation
+
+extension String {
+    var isNumber: Bool {
+        return !isEmpty && rangeOfCharacter(from: CharacterSet.decimalDigits.inverted) == nil
```
Comment (magnuskahr) [inline Sources/SwaggerSwiftCore/Utilities/String+IsNumber.swift:5]: quick tip, testing a string for `isNumber` can be done with `allSatisfy(\.isNumber)`, this addition of the empty check is good though

File: Sources/SwaggerSwiftCore/Models/NetworkRequestFunctionResponseType.swift [hunk 2]
```diff
@@ -139,6 +146,19 @@ case \(statusCode.rawValue):
         completion(.failure(.requestFailed(error: error)))
     }
 """
+        case .enumeration(let statusCode, let resultIsEnum, let responseType):
+            // This is necessary as iOS 12 doesnt support JSON fragments in JSONDecoder, so we have to do the parsing manually
+            return """
+            case \(statusCode.rawValue):
+                if let stringValue = String(data: data, encoding: .utf8) {
+                    // The string can be outputted as: "\\"enumValue\\"\\n", so we remove the newlines and remove the apostrophes
+                    let cleanedStringValue = stringValue.trimmingCharacters(in: CharacterSet.whitespacesAndNewlines).trimmingCharacters(in: CharacterSet(charactersIn: "\\""))
```
Comment (magnuskahr) [inline Sources/SwaggerSwiftCore/Models/NetworkRequestFunctionResponseType.swift:155]: The different may be minimal, but I think we should union the charactersets instead:
`CharacterSet.whitespacesAndNewlines.union(.init(charactersIn: "\\"))`

File: Sources/SwaggerSwiftCore/Utilities/String+IndentLines.swift [hunk 3]
```diff
@@ -0,0 +1,13 @@
+extension String {
+    func indentLines(_ count: Int) -> String {
+        self.split(separator: "\n", omittingEmptySubsequences: false)
+            .map {
+                if $0.trimmingCharacters(in: .whitespaces).count > 0 {
```
Comment (magnuskahr) [inline Sources/SwaggerSwiftCore/Utilities/String+IndentLines.swift:5]:
```suggestion
                if !$0.trimmingCharacters(in: .whitespaces).isEmpty {
```

File: Sources/SwaggerSwiftCore/Utilities/String+IsNumber.swift (hunk 1 above)
Comment (magnuskahr) [inline Sources/SwaggerSwiftCore/Utilities/String+IsNumber.swift:5]:
```suggestion
        return !isEmpty && allSatisfy(\.isNumber)
```

File: Sources/SwaggerSwiftCore/Models/NetworkRequestFunctionResponseType.swift (hunk 2 above)
Comment (magnuskahr) [inline Sources/SwaggerSwiftCore/Models/NetworkRequestFunctionResponseType.swift:155]:
```suggestion

                    let cleanedStringValue = stringValue.trimmingCharacters(in: CharacterSet.whitespacesAndNewlines.union(.init(charactersIn: "\\")))
```

**Review by magnuskahr — APPROVED** [review]
Approved with a few suggestions (also for code that was just moved)

**Pull request merged**